File: Python/8py 2/pythagorean_triple.py
"""
|--------------------------------------------------------------------------
| Pythagorean Triple
|--------------------------------------------------------------------------
|
"""

def pythagorean_triple(i):
   return sorted(i)[0]**2 + sorted(i)[1]**2 == sorted(i)[2]**2


def pythagorean_triple(integers):
   a, b, c = sorted(integers)
   return a * a + b * b == c * c


def pythagorean_triple(sides):
   return max(sides) ** 2 == sum(x**2 for x in sorted(sides)[:-1])


print(pythagorean_triple([3,4,5]), True)
print(pythagorean_triple([5,3,4]), True)
print(pythagorean_triple([13,12,5]), True)
print(pythagorean_triple([100,3,999]), False)

File: PHP/8php 2/pythagorean_triple.php
<?php
/*
|--------------------------------------------------------------------------
| Pythagorean Triple
|--------------------------------------------------------------------------
|
*/

function pythagorean_triple($int) {
   sort($int);
   return $int[0]**2 + $int[1]**2 == $int[2]**2;
}


function pythagorean_triple($int) {
   $int = array_map(function($v) {
      return pow($v, 2);
   }, $int);
   return max($int) === array_sum($int)/2 ? 1 : 0;
}


function pythagorean_triple($integers) {
   sort($integers);
   [$a2, $b2, $c2] = array_map(fn($i) => $i*$i, $integers);
   return $a2 + $b2 == $c2;
}


print_r(pythagorean_triple([3, 4, 5]) . ' ' . true . '
');
print_r(pythagorean_triple([3, 5, 9]) . ' ' . false . '
');